File: src/main/java/co/cubestudio/graphs/GraphNode.java
package co.cubestudio.graphs;

import java.util.ArrayList;
import java.util.List;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
class GraphNode {
    private final int id;
    private final String name;
    private final List<GraphArc> neighbors;
    private boolean flagged;

    GraphNode(int id, String name) {
        this.id = id;
        this.name = name;
        this.flagged = false;
        this.neighbors = new ArrayList<>();
    }

    void addArc(GraphArc arc) {
        neighbors.add(arc);
    }
}

File: src/main/typescript/co/cubestudio/graphs/GraphNode.ts
import { GraphArc } from './GraphArc';

export class GraphNode {
  private readonly _id: number;
  private readonly _name: string;
  private _neighbors: GraphArc[] = [];

  constructor(id: number, name: string) {
    this._id = id;
    this._name = name;
  }

  get id(): number {
    return this._id;
  }

  get name(): string {
    return this._name;
  }

  get neighbors(): GraphArc[] {
    return this._neighbors;
  }

  set neighbors(neighbors: GraphArc[]) {
    this._neighbors = neighbors;
  }

  addArc(arc: GraphArc): void {
    this._neighbors.push(arc);
  }
}
